src/main/scala/com/softwaremill/realworld/articles/ArticlesEndpoints.scala

```diff
@@ -16,19 +20,41 @@ object ArticlesEndpoints:
   given articleEncoder: zio.json.JsonEncoder[Article] = DeriveJsonEncoder.gen[Article]
   given articleDecoder: zio.json.JsonDecoder[Article] = DeriveJsonDecoder.gen[Article]
 
+  val articlesLayer: ZLayer[Any, Nothing, ArticlesService with DataSource] =
+    (DbContext.live >>> ArticlesRepository.live >>> ArticlesService.live) ++ DbContext.dbLayer
+
   // TODO add filtering
   // TODO add pagination
-  private val list: PublicEndpoint[Unit, String, List[Article], Any] = endpoint.get
+  val list: ZServerEndpoint[DataSource with ArticlesService, Any] = endpoint.get
```

Comment:
I'd pass dependencies using constructor parameters, instead of using the environment, as described here: https://zio.dev/reference/contextual/zio-environment-use-cases

Reply:
I've moved the deps to the constructor, but I'm unsure if that's what you meant. Please take a look.